js/pages/incidence-rates/components/iranalysis/components/results.js

```diff
@@ -73,15 +73,23 @@ define([
 				return "per " + multiplier  + " persons";
 			});
 
+			this.selectedSourceId = sharedState.IRAnalysis.selectedSourceId;
+
 			// observable subscriptions
 
 			this.subscriptions.push(this.selectedTarget.subscribe((newVal) => {
-				if (this.selectedSource()) // this will cause a report refresh
+				if (this.selectedSource()) {// this will cause a report refresh
 					this.selectSource(this.selectedSource());
+				} else {
+					this.selectSourceBySourceId();
+				}
 			}));
 			this.subscriptions.push(this.selectedOutcome.subscribe((newVal) => {
-				if (this.selectedSource()) // this will cause a report refresh
+				if (this.selectedSource()) {// this will cause a report refresh
 					this.selectSource(this.selectedSource());
+				} else {
+					this.selectSourceBySourceId();
```

Comment:
Would be better to avoid `if` here and have single path

Reply:
fixed